### ISSUE TITLE
add better documentation on configuring ncmpcpp visualizer 

### ISSUE THREAD

**Issue by verajosemanuel** [mopidy/mopidy#1985]
After some research I could not find any clue on configuring the ncmpcpp visualizer.

All I got this issue mentioning a better support for mopidy 

https://github.com/mopidy/mopidy/issues/775

some hints here and there, but not a thorough explain

thanks

**Comment by gnmearacaun** [comment]
There is an example here: https://github.com/ncmpcpp/ncmpcpp/blob/35f9262a6e7c7321f6cfd9a55a1d7db920b81b22/doc/config#L66
Typically you will uncomment (remove the `#` at the beginning of) the lines you want to be activated. Perhaps confusingly, commented lines in the mopidy config appear to represent the defaults, and only may need to be uncommented if you want to alter those same defaults.

**Comment by gnmearacaun** [comment]
The ncmpcpp example configuration file is also located in /usr/share/doc/ncmpcpp on debian-based systems. A line is included for mopidy.conf to help setup visualization. If you're running mopidy as a systemd service with `sudo systemctl enable mopidy` you may want to run `sudo adduser mopidy audio` and `sudo chown mopidy /etc/mopidy/mopidy.conf` if you run into permission problems.

**Comment by kingosticks** [comment]
> Perhaps confusingly, commented lines in the mopidy config appear to represent the defaults, and only may need to be uncommented for the purpose of changing those same defaults.

Can you clarify what you consider confusing? Config files supporting comments are very common. Programs that ship a config file containing all the defaults commented out is also very common - but we don't actually do that with our _service_, we ship an [empty config file](https://salsa.debian.org/mopidy-team/mopidy/-/blob/debian/master/debian/etc/mopidy.conf). We also [explicitly suggest people do not add stuff they don't need to their Mopidy config file](https://docs.mopidy.com/en/latest/config/#editing-the-configuration).

If you run Mopidy [in the terminal](https://docs.mopidy.com/en/latest/running/terminal/) (where the config file lives in your home directory, i.e. NOT the service), Mopidy will generate a config file for you on first-run if there isn't one already present and that generated file will contain the default config values all commented out. I guess this is what you are talking about. I would imagine (and hope) most people are using the service so they don't get this.

> you may want to run `sudo adduser mopidy audio`, `sudo adduser mopidy sudo` and `sudo chown mopidy /etc/mopidy/mopidy.conf` if you run into permission problems. 

When the Mopidy service is installed it will add the mopidy user to the audio group and set the owner of /etc/mopidy/mopidy.conf as per [here](https://salsa.debian.org/mopidy-team/mopidy/-/blob/debian/master/debian/mopidy.postinst#L12-22). Why would you want to do `sudo adduser mopidy sudo`? That makes no sense to me.

Appreciate you taking the time to do a write-up but FYI, from your blog post:

> When configuring extensions in our ~/.config/mopidy/mopidy.conf file, we need to restart the service for edits to take effect.

> The current state of the server depends on a blend of /etc/mopidy/mopidy.conf and that of our local ~/.config/mopidy/mopidy.conf 

No to all of this. The [service config](https://docs.mopidy.com/en/latest/running/service/#configuration) lives in /etc/mopidy/mopidy.conf, the Mopidy service does not read ~/.config/mopidy/mopidy.conf/ ever. That goes for the other place(s) you reference ~/.config/mopidy/mopidy.config.

**Comment by kingosticks** [comment]
And on topic, if someone wants to add a new bit to our "Advanced Setups" for ncmpcpp's visualizer they are very welcome to submit a PR.

**Comment by gnmearacaun** [comment]
Yes, that's what was confusing me in retrospect. The generated .config/mopidy/mopidy.conf with the defaults commented out is something I was not quite accustomed to (as opposed to a config file containing comments). 
 I think the permissions problem caused me to try some reckless commands. Whereas it says in the [docs](https://docs.mopidy.com/en/latest/running/service/) `mopidy` will need read access to play any music, it doesn't suggest which command to do so. 
I have amended the info on my blog in a new section about local music. I'm still foggy on a couple of points, and I will address further when I am clear about it.

**Comment by kingosticks** [comment]
No worries, I just wanted to avoid going too far off the original topic. Please do ask any other questions on the forum.

**Comment by gnmearacaun** [comment]
I realized my trouble was because I had removed the contents of the original `/etc/mopidy/mopidy.conf` and replaced it with the entire `~/.config/mopidy/mopidy.conf` later slimming that down, but missing the [core] and [logging] sections. Not too smart, lol!

I reinstalled everything and configured the service for my debian system as per [the docs](https://docs.mopidy.com/en/latest/running/service/#service-management-on-debian) 

Now scanning (and visualization) works fine without the permissions hacks, and my blogpost became less complicated. 
 
>  When configuring extensions in our ~/.config/mopidy/mopidy.conf file, we need to restart the service for edits to take effect.

> No to all of this

@kingosticks  It seemed I needed to restart the service after changes to /etc/mopidy/mopidy.conf for changes to take effect, can you clarify?

**Comment by kingosticks** [comment]
You are correct, the config file is only read when Mopidy starts and so you must restart Mopidy for any config change to take effect. The "all" there was because I was referring to all mentions of "~/.config/mopidy/mopidy.conf" in what I had quoted.

**Comment by gnmearacaun** [comment]
Understood. Thanks!

**Comment by alamoudimoh** [comment]
Again, th3 documents from the point of the developers view. I am not sure how manay people would need to report that documents are confusing to believe.


FYI,,, it is not about you, it is about ur users

**Comment by kingosticks** [comment]
@alamoudimoh I don't consider your comments on these issues constructive or helpful. It's approaching spam now. Please stop.

**Comment by alamoudimoh** [comment]
> @alamoudimoh I don't consider your comments on these issues constructive or helpful. It's approaching spam now. Please stop.

Apologies if i was understood wrongly, this isn't my attention, i like the solution, i really do, have been trying to see it for a year now, but with no luck. Nothing comprehensive here nor on YouTube, i mentioned the same comment to a YouTuber how just gave a brief how to cuz he doesn't want the videos to be long. Simpl3 create more videos, 

I just wanted to highlight a point, it is somehow lavk of clear documentation.

**Comment by lissethamc** [comment]
Hi! I’m reading through this issue and the discussion, and I was wondering whether a documentation contribution around this topic would still be useful, given that the issue has been open for a while.

I’m currently getting familiar with the Mopidy repository and documentation, but I’d be happy to look into this and see if a small documentation improvement would still make sense here.

**Comment by jayyycodes** [comment]
Hi, is this still relevant? I'd like to work on it if so.
